[ILM] Fix hot phase serialization

alisonelizabeth opened the pull request:
Fixes https://github.com/elastic/kibana/issues/87208

This looks like it might be a regression of https://github.com/elastic/kibana/pull/85579, which is part of the 7.11 work, so skipping release note.

**How to test:**
1. Go to ILM and create a policy.
2. Open advanced settings for the hot phase and enable the force merge action. Specify number of segments and enable the "Compress stored fields" toggle.
3. Enable the "readonly" toggle for the hot phase.
4. Verify the JSON preview and save policy.
5. Reopen policy and verify settings saved correctly.
6. Verify the same behavior in steps 1-5, but with "Use recommended defaults" disabled.
7. Verify behavior is as expected when "Enable rollover" is disabled. (From what I can tell, all fields should be hidden except "Index priority").

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/96740/)
* Commit: 1af213c580f680a259b4922429b2ac12452393d9

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/9c092d5264e91df9054f2883cee92b6790a4a718) | [after](https://github.com/elastic/kibana/commit/1af213c580f680a259b4922429b2ac12452393d9) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 249.7KB | 249.7KB | +5.0B |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/96740/","number":"96740","commit":"1af213c580f680a259b4922429b2ac12452393d9"}]}
PIPELINE-->

jloleysens commented:
@elasticmachine merge upstream

kibanamachine commented:
ignoring request to update branch, pull request is closed

jloleysens commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97211/)
* Commit: ecee97c4ecbb1a85b0412891f7dd55d0d4b0b874

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/e63e3d869d2827bb8e2c6dffb1ba02634dea4cc2) | [after](https://github.com/elastic/kibana/commit/ecee97c4ecbb1a85b0412891f7dd55d0d4b0b874) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 249.7KB | 249.7KB | +5.0B |

### History
* :green_heart: [Build #96740](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/96740/) succeeded 1af213c580f680a259b4922429b2ac12452393d9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/96740/","number":"96740","commit":"1af213c580f680a259b4922429b2ac12452393d9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97211/","number":"97211","commit":"ecee97c4ecbb1a85b0412891f7dd55d0d4b0b874"}]}
PIPELINE-->

alisonelizabeth commented:
@jloleysens would you mind taking another look at this when you have time? I incorporated your changes from https://github.com/alisonelizabeth/kibana/pull/3. I also addressed the bugs I mentioned in https://github.com/alisonelizabeth/kibana/pull/3#issuecomment-755321254 via https://github.com/elastic/kibana/pull/87213/commits/bfcdfb4c469cc4347bdc865ecd9ce8694bb263e2. Thanks!

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97369/)
* Commit: 206016a649bf89d66839273ca92a3732431d4c3c

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/a95fdbdec3b83363e3416a3bbe5e62faf49694eb) | [after](https://github.com/elastic/kibana/commit/206016a649bf89d66839273ca92a3732431d4c3c) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 249.8KB | 250.1KB | +321.0B |

#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/a95fdbdec3b83363e3416a3bbe5e62faf49694eb) | [after](https://github.com/elastic/kibana/commit/206016a649bf89d66839273ca92a3732431d4c3c) | diff |
| --- | --- | --- | --- |
| `default` | 47256 | 48019 | +763 |

### History
* :green_heart: [Build #97211](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97211/) succeeded ecee97c4ecbb1a85b0412891f7dd55d0d4b0b874
* :green_heart: [Build #96740](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/96740/) succeeded 1af213c580f680a259b4922429b2ac12452393d9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/96740/","number":"96740","commit":"1af213c580f680a259b4922429b2ac12452393d9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97211/","number":"97211","commit":"ecee97c4ecbb1a85b0412891f7dd55d0d4b0b874"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97369/","number":"97369","commit":"206016a649bf89d66839273ca92a3732431d4c3c"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97384/)
* Commit: bfcdfb4c469cc4347bdc865ecd9ce8694bb263e2

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/a95fdbdec3b83363e3416a3bbe5e62faf49694eb) | [after](https://github.com/elastic/kibana/commit/bfcdfb4c469cc4347bdc865ecd9ce8694bb263e2) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 249.8KB | 250.1KB | +358.0B |

#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/a95fdbdec3b83363e3416a3bbe5e62faf49694eb) | [after](https://github.com/elastic/kibana/commit/bfcdfb4c469cc4347bdc865ecd9ce8694bb263e2) | diff |
| --- | --- | --- | --- |
| `default` | 47256 | 48019 | +763 |

### History
* :green_heart: [Build #97369](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97369/) succeeded 206016a649bf89d66839273ca92a3732431d4c3c
* :green_heart: [Build #97211](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97211/) succeeded ecee97c4ecbb1a85b0412891f7dd55d0d4b0b874
* :green_heart: [Build #96740](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/96740/) succeeded 1af213c580f680a259b4922429b2ac12452393d9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/96740/","number":"96740","commit":"1af213c580f680a259b4922429b2ac12452393d9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97211/","number":"97211","commit":"ecee97c4ecbb1a85b0412891f7dd55d0d4b0b874"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97369/","number":"97369","commit":"206016a649bf89d66839273ca92a3732431d4c3c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97384/","number":"97384","commit":"bfcdfb4c469cc4347bdc865ecd9ce8694bb263e2"}]}
PIPELINE-->

jloleysens commented:
@elasticmachine merge upstream

jloleysens reviewed (approved):
Tested locally and the changes look good to me! Great work @alisonelizabeth 🍻

I think it would be really awesome to add a test case for the first scenario you outlined here: https://github.com/alisonelizabeth/kibana/pull/3#issuecomment-755321254. To ensure that hidden rollover fields do not block form submission - not blocking on this though!

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97497/)
* Commit: e4673092a95da0f874f9f026558cc5e3c5f1109b

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ba20a74be39701061aa5fa6f57483315c3de3c0b) | [after](https://github.com/elastic/kibana/commit/e4673092a95da0f874f9f026558cc5e3c5f1109b) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 249.8KB | 250.1KB | +358.0B |

### History
* :green_heart: [Build #97384](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97384/) succeeded bfcdfb4c469cc4347bdc865ecd9ce8694bb263e2
* :green_heart: [Build #97369](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97369/) succeeded 206016a649bf89d66839273ca92a3732431d4c3c
* :green_heart: [Build #97211](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97211/) succeeded ecee97c4ecbb1a85b0412891f7dd55d0d4b0b874
* :green_heart: [Build #96740](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/96740/) succeeded 1af213c580f680a259b4922429b2ac12452393d9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/96740/","number":"96740","commit":"1af213c580f680a259b4922429b2ac12452393d9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97211/","number":"97211","commit":"ecee97c4ecbb1a85b0412891f7dd55d0d4b0b874"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97369/","number":"97369","commit":"206016a649bf89d66839273ca92a3732431d4c3c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97384/","number":"97384","commit":"bfcdfb4c469cc4347bdc865ecd9ce8694bb263e2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97497/","number":"97497","commit":"e4673092a95da0f874f9f026558cc5e3c5f1109b"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97610/)
* Commit: 7ad2547b59649c2ca17cc0dafe712145085d7f9a

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/91aed6f961191a607742540657a230d04a7202a8) | [after](https://github.com/elastic/kibana/commit/7ad2547b59649c2ca17cc0dafe712145085d7f9a) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 249.9KB | 250.2KB | +358.0B |

### History
* :green_heart: [Build #97497](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97497/) succeeded e4673092a95da0f874f9f026558cc5e3c5f1109b
* :green_heart: [Build #97384](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97384/) succeeded bfcdfb4c469cc4347bdc865ecd9ce8694bb263e2
* :green_heart: [Build #97369](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97369/) succeeded 206016a649bf89d66839273ca92a3732431d4c3c
* :green_heart: [Build #97211](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97211/) succeeded ecee97c4ecbb1a85b0412891f7dd55d0d4b0b874
* :green_heart: [Build #96740](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/96740/) succeeded 1af213c580f680a259b4922429b2ac12452393d9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/96740/","number":"96740","commit":"1af213c580f680a259b4922429b2ac12452393d9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97211/","number":"97211","commit":"ecee97c4ecbb1a85b0412891f7dd55d0d4b0b874"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97369/","number":"97369","commit":"206016a649bf89d66839273ca92a3732431d4c3c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97384/","number":"97384","commit":"bfcdfb4c469cc4347bdc865ecd9ce8694bb263e2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97497/","number":"97497","commit":"e4673092a95da0f874f9f026558cc5e3c5f1109b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/97610/","number":"97610","commit":"7ad2547b59649c2ca17cc0dafe712145085d7f9a"}]}
PIPELINE-->
